src/legacy/core_plugins/metrics/server/lib/vis_data/helpers/bucket_transform.js

```diff
@@ -78,11 +78,13 @@ export const bucketTransform = {
   },
   static: bucket => {
     checkMetric(bucket, ['value']);
+    // Anything containing a decimal point or an exponent is considered decimal value
+    const isDecimalValue = bucket.value.match(/[.e]/i);
```

Comment:
👍

Reply:
As long as we're inside JS I think checking for `null` as a falsey value is fine, but I can wrap it. I cannot convert it to a number, because of the issue you pointed out that it will actually create the 64 bit rounding issues for the check, but then when we later write the script the string value won't have those, so `Number('1231123112631231233.123').toString() == "1231123112631231200"` meaning we will append a `L` and then Elasticsearch will fail because it's not a valid long.